spec/classes/init_spec.rb

```diff
@@ -25,6 +25,14 @@
           it { is_expected.to contain_apt__source('gitlab_official_ee').with_ensure('absent') }
           it { is_expected.not_to contain_apt__source('gitlab_official_') }
           it { is_expected.not_to contain_yumrepo('gitlab_official_ce') }
+          case facts[:operatingsystem]
```

Comment:
@jkroepke if you get a chance can you submit a pull that uses modern fact for this?

Reply:
Last time I'm writing spec test, modern facts are simply not work or not defined by the module you are using for generated the facts. 

No idea whats the current state is.

Reply:
For information :
structured facts with rspec-puppet : `facts[:os]['family']`
structured facts with beaker-rspec : `fact('os.family')`